src/main/java/practice/StreamPractice.java

```diff
@@ -14,57 +14,54 @@
 
     public int findMinEvenNumber(List<String> numbers) {
         return numbers.stream()
-                .map(s -> s.split(","))
-                .flatMap(Arrays::stream)
+                .flatMap(s -> Arrays.stream(s.split(",")))
                 .map(Integer::parseInt)
                 .filter(n -> n % 2 == 0)
                 .min(Integer::compare)
-                .orElseThrow(() ->
-                        new RuntimeException("Can't get min value from list: "
-                                + numbers));
-
+                .orElseThrow(() -> new RuntimeException("Can't get min value from list: "
+                        + numbers));
     }
 
     public Double getOddNumsAverage(List<Integer> numbers) {
         return IntStream.range(0, numbers.size())
                 .map(n -> n % 2 == 1 ? numbers.get(n) - 1 : numbers.get(n))
                 .filter(n -> n % 2 == 1)
                 .average()
-                .orElseThrow(() ->
-                        new NoSuchElementException("Can't do this for list: "
-                                + numbers));
-
+                .getAsDouble();
     }
 
     public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toAge) {
+        Predicate<Person> validMan = p -> p.getSex() == Person.Sex.MAN
+                && p.getAge() >= fromAge
+                && p.getAge() <= toAge;
         return peopleList.stream()
-                .filter(p -> p.getSex() == Person.Sex.MAN
-                        && p.getAge() >= fromAge
-                        && p.getAge() <= toAge)
+                .filter(validMan)
                 .collect(Collectors.toList());
     }
 
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
+        Predicate<Person> validPeoples = p -> p.getAge() >= fromAge
+                && (p.getSex() == Person.Sex.MAN && p.getAge() <= maleToAge
+                || p.getSex() == Person.Sex.WOMAN && p.getAge() <= femaleToAge);
         return peopleList.stream()
-                .filter(p -> p.getAge() >= fromAge
-                        && p.getSex() == Person.Sex.MAN && p.getAge() <= maleToAge
-                        || p.getSex() == Person.Sex.WOMAN && p.getAge() <= femaleToAge)
+                .filter(validPeoples)
                 .collect(Collectors.toList());
     }
 
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
+        Predicate<Person> validWomen = p -> p.getSex() == Person.Sex.WOMAN
+                && p.getAge() >= femaleAge;
         return peopleList.stream()
-                .filter(p -> p.getSex() == Person.Sex.WOMAN
-                        && p.getAge() >= femaleAge)
+                .filter(validWomen)
                 .map(Person::getCats)
                 .flatMap(Collection::stream)
```

Comment:
Я думаю, тут лучше сократить, чтобы уменьшить количество итераций по коллекции.
```suggestion
.flatMap(x -> x.getCats().stream())
```

Reply:
done